### DISCUSSION TITLE
Add icon support to the tabs component

### DISCUSSION THREAD

**Discussion by HiDeoo** [withastro/starlight#1453] [Feature Requests]
### What version of `starlight` are you using?

0.17.2

### What is your idea?

Adding a new `icon` prop to the `<TabItem>` component that would optionally accept an icon [`name`](https://starlight.astro.build/guides/components/#all-icons) from the Starlight icon set and render it next to the tab label.

### Why is this feature necessary?

This would allow for more flexibility in the design of tabbed interfaces and make it easier to visually distinguish between tabs.

The idea originally comes from a features from Erika on the [`starlight-package-managers` component](https://github.com/HiDeoo/starlight-package-managers/issues/5) (which internally uses the default Starlight tabs and only enhances them with sync between the tabs) to display package manager icons next to the package manager names.

### Do you have examples of this feature in other projects?

The Biome website [uses](https://biomejs.dev/guides/getting-started/#installation) that approach for package managers related comnmands in their documentation and I personally think it looks great.

<img width="738" alt="image" src="https://github.com/withastro/starlight/assets/494699/c8ba9cc2-d381-4dc6-9b7d-24fa01218404">

For the specific use case of `starlight-package-managers`, https://github.com/withastro/starlight/pull/1308 already adds icons for `npm`, `yarn`, and `pnpm`. `bun` could be added as well at some point if this is something of interest.

### Participation

- [X] I am willing to submit a pull request for this feature.

**Comment by delucis** [comment]
Sorry for the slow response on this one — yes I don’t see why we shouldn’t support this. And adds an argument for exposing all those extra icons in #1308 😁 

The one difference from the above example I guess is that these would be monochrome icons, but I guess that’s OK?

**Reply by HiDeoo** [reply]
No worries at all. Thanks for checking out the idea.

> The one difference from the above example I guess is that these would be monochrome icons, but I guess that’s OK?

Personally, I think that is more than fine. I did not even add the screenshot to show what the user should be able to do exactly, but more like the user should be able to do something similar with the default styles and icons provided by Starlight.

I'll start playing with this when I get some time and submit a PR 👍

**Comment by HiDeoo** [comment]
Available in Starlight [`v0.21.0`](https://github.com/withastro/starlight/releases/tag/@astrojs/starlight@0.21.0) using the new [`icon` attribute](https://starlight.astro.build/guides/components/#tabs) for the `<TabItem>` component.
